modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/clientExtensionFilter.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {ClientExtension} from 'frontend-js-components-web';
+import PropTypes from 'prop-types';
+import React from "react";
+
+const getSelectedItemsLabel =
+	(filterProps) =>
+		filterProps.cxFilterImpl.getSelectedItemsLabel(filterProps);
+const getOdataString =
+	(filterProps) => filterProps.cxFilterImpl.getOdataString(filterProps);
+
+function ClientExtensionFilter({
+	cxFilterImpl,
+	selectedData,
+	setFilter,
+}) {
+	return (
+		<ClientExtension
+			args={{
+				filter: {
+					selectedData
+				},
+				setFilter: ({
```

Comment:
This is for sandboxing CX invocations of `setFilter`, so that they cannot pass in extra data that may alter FDS internal status.

Reply:
I'm in doubt about hiding the `active` prop in `setFilter()`. Reason is that a CX filter may want to disable itself. We have this interaction in the ootb filters ([example](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DateRangeFilter.js#L159)), as the filter UX allows to edit/delete it.

Once `setFilter()` can work without the `id`, which becomes a property set by whoever configures the FDS (and no longer the CX) is there any reason to restrict this function, that I'm not seeing?

Reply:
Nope, I hardcoded it but thought that it could be made configurable too. In fact, if you want to add some button to clear the filter you would need it, most probably. :+1: